src/List.js

```diff
@@ -140,8 +140,12 @@ export class List extends IndexedCollection {
     return mergeIntoListWith(this, deepMerger, arguments);
   }
 
-  mergeDeepWith(merger, ...iters) {
-    return mergeIntoListWith(this, deepMergerWith(merger), iters);
+  mergeDeepWith(...args) {
+    var keyPath, merger, iters;
+    typeof args[0] === 'function' ?
+      [merger, ...iters] = args :
+      [keyPath, merger, ...iters] = args;
```

Comment:
I think providing a keyPath as a first argument is confusing. When would someone write code that uses it? I fear that it's more likely that someone accidentally hits this codepath by forgetting to provide a `merger` function.